library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -1101,4 +1102,16 @@ private int translateStatus(int status) {
             }
         }
     }
+
+    /**
+     * Restart will activate the downloads workflow in case that it was not already active
+     * <p/>
+     * A possible scenario: A client denies a download for a particular business rule and that
+     * rule does not apply any more. Calling this method will reactivate the downloads workflow,
+     * check the client rules and proceed if necessary
+     */
+    public void restart() {
```

Comment:
The downloadmanager has some mechanisms in order to restart itself automatically. For example when a new row is inserted in the database or there is a connection change, but there are cases where the client business rules don't modify the database and there is no network change either.

For example, let's say that in order to download something the user needs to be registered. If the user is not the download cannot proceed. Once the user is registered, the download can proceed.

In this scenario, after the user has registered, we can call `restart`, the download manager will proceed with all the checks and this time the paused download will be resumed as the client no longer denies.

But it could be any other random reason.